presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorWithEstimatedExchanges.java

```diff
@@ -189,6 +199,14 @@ public static LocalCostEstimate calculateRemoteRepartitionCost(double inputSizeI
         return LocalCostEstimate.of(inputSizeInBytes, 0, inputSizeInBytes);
     }
 
+    public static LocalCostEstimate calculateCteProducerCost(Session session, StatsProvider statsProvider, PlanNode source)
+    {
+        double inputSizeInBytes = statsProvider.getStats(source).getOutputSizeInBytes(source);
+        // default HDFS replication
+        double cteHdfsReplicationFactor = session.getSystemProperty(DEFAULT_CTE_PRODUCER_REPLICATION_COEFFICIENT, Double.class);
```

Comment:
Why not add this as a function in SystemSessionProperties like other properties?